### ISSUE TITLE
Add ability to specify Access Key ID and Secret Key in setup endpoint

### ISSUE THREAD

**Issue by keimoon** [treeverse/lakeFS#908]
We need this for integrating lakeFS with CI: an instance of lakeFS will be created before a build is run and be destroyed after that. Every time lakeFS is set up, it will create different set up Access Key ID / Secret Key so it is hard to inject those values to the test.

**Comment by arielshaqed** [comment]
Could you expand on the use-case, please?  I'm curious because I am not aware of many systems that allow injecting secrets into new users.  E.g. with AWS I usually face a similar difficulty: either use pre-existing credentials or create new ones, extract those and use them.

We actually have code that does something similar in lakeFS.  It runs as part of our system test "nessie".  [`testutil.SetupTestingEnv`](https://github.com/treeverse/lakeFS/blob/e5563395db6795e986f931793d582ab0eb992392/testutil/setup.go#L28) shows how to create the first admin user and extract its credentials.

**Comment by keimoon** [comment]
The way our CI is set up is to create database resources (postgres, ... ) first, then running tests in parallel later. All created database instances are bounded to a build, which is independent to each other. We also make sure all tests in a build are independent, stateless and should not rely on any initialization system, so configurations and credentials should be pre-determined. Other databases we are using (postgres, redis, ...) support some API to setup an account with pre-determined credential, while lakeFS does not. Of course I can use a hack to inject a key to lakeFS's postgres directly, but it is nice to have a higher level API instead.

**Comment by arielshaqed** [comment]
Thanks for the clearly explaining a use-case!  I usually think of it as an S3-like system, which does not offer this ability.
We would like to try to add this ability as an option to the existing command `lakefs superuser` (_not_ `lakectl`, but `lakefs`...).   Currently it creates access key ID and secret key, but we could inject those.
Note that some clients for AWS S3 (entirely unrelated to lakefs) might decide to validate access keys and secrets and enforce a particular format for them.  Such clients might enforce particular formats for access keys and secrets.

**Comment by arielshaqed** [comment]
@keimoon also if you would prefer a more chat-based discussion format for questions that may come up, please [join our community Slack](https://join.slack.com/t/lakefs/shared_invite/zt-g86mkroy-186GzaxR4xOar1i1Us0bzw).  (Of course issues are a great way to ask for new features!)
